homeassistant/components/ozw/websocket_api.py

```diff
@@ -15,101 +18,159 @@
+    """Get the statistics for a Z-Wave node."""
+    manager = hass.data[DOMAIN][MANAGER]
+    stats = (
+        manager.get_instance(msg[OZW_INSTANCE]).get_node(msg[NODE_ID]).get_statistics()
     )
-    def websocket_node_statistics(self, hass, connection, msg):
-        """Get the statistics for a Z-Wave node."""
-
-        stats = (
-            self._manager.get_instance(msg[OZW_INSTANCE])
-            .get_node(msg[NODE_ID])
-            .get_statistics()
-        )
-        connection.send_result(
-            msg[ID],
-            {
-                "node_id": msg[NODE_ID],
-                "send_count": stats.send_count,
-                "sent_failed": stats.sent_failed,
-                "retries": stats.retries,
-                "last_request_rtt": stats.last_request_rtt,
-                "last_response_rtt": stats.last_response_rtt,
-                "average_request_rtt": stats.average_request_rtt,
-                "average_response_rtt": stats.average_response_rtt,
-                "received_packets": stats.received_packets,
-                "received_dup_packets": stats.received_dup_packets,
-                "received_unsolicited": stats.received_unsolicited,
-                OZW_INSTANCE: msg[OZW_INSTANCE],
-            },
-        )
+    connection.send_result(
+        msg[ID],
+        {
+            "node_id": msg[NODE_ID],
+            "send_count": stats.send_count,
+            "sent_failed": stats.sent_failed,
+            "retries": stats.retries,
+            "last_request_rtt": stats.last_request_rtt,
+            "last_response_rtt": stats.last_response_rtt,
+            "average_request_rtt": stats.average_request_rtt,
+            "average_response_rtt": stats.average_response_rtt,
+            "received_packets": stats.received_packets,
+            "received_dup_packets": stats.received_dup_packets,
+            "received_unsolicited": stats.received_unsolicited,
+            OZW_INSTANCE: msg[OZW_INSTANCE],
+        },
+    )
+
+
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "ozw/refresh_node_info",
+        vol.Optional(OZW_INSTANCE, default=1): vol.Coerce(int),
+        vol.Required(NODE_ID): vol.Coerce(int),
+    }
+)
+def websocket_refresh_node_info(hass, connection, msg):
+    """Tell OpenZWave to re-interview a node."""
+
+    manager = hass.data[DOMAIN][MANAGER]
+    options = hass.data[DOMAIN][OPTIONS]
+
+    @callback
+    def forward_node(node):
+        """Forward node events to websocket."""
+        if node.node_id != msg[NODE_ID]:
+            return
+
+        forward_data = {
+            "type": "node_updated",
+            "node_query_stage": node.node_query_stage,
+        }
+        connection.send_message(websocket_api.event_message(msg["id"], forward_data))
+
+    @callback
+    def async_cleanup() -> None:
+        """Remove signal listeners."""
+        for unsub in unsubs:
+            unsub()
```

Comment:
Should we test the unsubscribe too? We can unsubscribe and then receive an MQTT message and check that we didn't generate a websocket message.